lib/AST/AutoDiff.cpp

```diff
@@ -67,6 +67,58 @@ autodiff::getNumAutoDiffAssociatedFunctions(unsigned differentiationOrder) {
   return differentiationOrder * 2;
 }
 
+bool autodiff::getBuiltinAutoDiffApplyConfig(
+    StringRef operationName, AutoDiffAssociatedFunctionKind &kind,
+    unsigned &arity, unsigned &order, bool &rethrows, bool &isMethod) {
+  // SWIFT_ENABLE_TENSORFLOW
+  if (!operationName.startswith("autodiffApply_"))
+    return false;
+  operationName = operationName.drop_front(strlen("autodiffApply_"));
+  // Parse 'jvp' or 'vjp'.
+  if (operationName.startswith("jvp"))
+    kind = AutoDiffAssociatedFunctionKind::JVP;
+  else if (operationName.startswith("vjp"))
+    kind = AutoDiffAssociatedFunctionKind::VJP;
```

Comment:
an `else` `return false` branch would be good here, to indicate the intention that jvp/vjp is mandatory.

Reply:
will do that in my next patch.